cloudify_types/cloudify_types/component/component_operations.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2017-2019 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from cloudify.decorators import operation
+
+import utils
+from .component import Component
+
```

Comment:
So I was thinking that it would be best to take out the external resource logic, as I mentioned in upload blueprint/create deployment. And put it in each of these operations.  So we do:
```python
component = Component(_)
if not utils.external_resource:
    return component.upload()
```
Obv. there are a few ways to play with this. Possibly things can be rearranged for it to fit into `utils. proxy_operation`.